specs/bali/Profile_Properties_Definition.md

```diff
@@ -0,0 +1,246 @@
+
+* ProvisioningPropertiesSpec
+```go
+type ProvisioningPropertiesSpec struct {
+	// DataStorage represents some suggested data storage capabilities.
+	DataStorage DataStorageLoS `json:"dataStorage,omitempty"`
+	// IOConnectivity represents some suggested IO connectivity capabilities.
+	IOConnectivity IOConnectivityLoS `json:"ioConnectivity,omitempty"`
+}
+```
+
+* ReplicationPropertiesSpec
+```go
+type ReplicationPropertiesSpec struct {
+	// DataProtection represents some suggested data protection capabilities.
+	DataProtection DataProtectionLoS `json:"dataProrection,omitempty"`
+	// ReplicaInfos represents some suggested data replicaiton information.
+	ReplicaInfos struct {
+		// The enumeration literal specifies whether the target elements will be
+		// updated synchronously or asynchronously. The possible values of this
+		// property could be:
+		// * Active: Active-Active (i.e. bidirectional) synchronous updates.
+		// * Adaptive: Allows implementation to switch between synchronous
+		//   and asynchronous modes.
+		// * Asynchronous: Asynchronous updates.
+		// * Synchronous: Synchronous updates.
+		ReplicaUpdateMode string `json:"replicaUpdateMode,omitempty"`
+		// ConsistencyEnabled indicates that the source and target shall be
+		// consistent. The default value is false.
+		ConsistencyEnabled bool `json:"consistencyEnabled,omitempty"`
+		// ReplicationPeriod shall be an ISO 8601 duration that defines the
+		// duration of performing replication operation. For example,
+		// "P3Y6M4DT12H30M5S" represents a duration of "3 years, 6 months,
+		// 4 days, 12 hours, 30 minutes and 5 seconds".
+		ReplicationPeriod string `json:"replicationPeriod,omitempty"`
+		// ReplicationBandwidth specifies the maximum bandwidth for performing
+		// replication operation.
+		// +units:Mb/s
+		ReplicationBandwidth int64 `json:"replicationBandwidth,omitempty"`
+	} `json:"replicaInfos,omitempty"`
+}
+```
+
+* SnapshotPropertiesSpec
+```go
+type SnapshotPropertiesSpec struct {
+	// The property defines how to take snapshots.
+	Schedule struct {
+		// This vaule is represented as a string in ISO 8601 datetime format.
+		// ISO 8601 sets out an internationally agreed way to represent dates:
+		// yyyy-mm-ddThh:mm:ss.ffffff. For example, "3:53pm, September 15, 2008"
+		// is represented as "2008-09-15T15:53:00".
+		Datetime string `json:"datetime,omitempty"`
+		// The value specifies the duration of executing a operation, which
+		// contains three options including Daily, Weekly and Monthly.
+		Occurrence string `json:"occurrence,omitempty"`
+	} `json:"schedule,omitempty"`
+	Retention struct {
+		// The value specifies the number of recently created snapshots for
+		// retention.
+		// +optional
+		Number int64 `json:"number,omitempty"`
+		// The value specifies the duration of snapshots for retention.
+		// +optional
+		// +units:day
+		Duration int64 `json:"duration,omitempty"`
```

Comment:
If both Number and Duration are specified, which one should be used?

Reply:
I prefer that the system will apply these two rules both, but I'm not sure if it's difficult to implement that in code.

Reply:
If both are used, one will override the other.

Reply:
Then I guess Number should be considered as first citizen.